package dmModelDDF;

import "ddf/ddf_extensions.proto";
import "ddf/ddf_math.proto";

option java_package = "com.dynamo.model.proto";
option java_outer_classname = "Model";
option (ddf_java_package) = "com.dynamo.model.ddf";

message ModelDesc
{
    required string Name = 1;
    required string Mesh = 2 [(resource)=true];
    required string Material = 3 [(resource)=true];
    repeated string Textures = 4 [(resource)=true];
}

message SetVertexConstant
{
    required uint32 Register = 1;
    required dmMath.Vector4 Value = 2;
}

message ResetVertexConstant
{
    required uint32 Register = 1;
}

message SetFragmentConstant
{
    required uint32 Register = 1;
    required dmMath.Vector4 Value = 2;
}

message ResetFragmentConstant
{
    required uint32 Register = 1;
}

message SetTexture
{
    required uint64 TextureHash = 1;
    required uint32 TextureUnit = 2;
}
